TMDS_3/include/CGAL/Mesh_complex_3_in_triangulation_3.h

```diff
@@ -0,0 +1,2075 @@
+  std::size_t tbb_hasher(const std::pair<CGAL::internal::CC_iterator<DSC, Const>,
+    CGAL::internal::CC_iterator<DSC, Const> >& p)
+  {
+    return boost::hash<std::pair<CGAL::internal::CC_iterator<DSC, Const>,
+      CGAL::internal::CC_iterator<DSC, Const> > >()(p);
+  }
+}//end namespace CGAL
+#endif
+
+namespace CGAL {
+
+  namespace TMDS_3 {
+
+    namespace details {
+
+      template <typename Tr>
+      class C3t3_helper_class
+      {
+      protected:
+        typedef typename Tr::Vertex_handle Vertex_handle;
+        typedef typename Tr::Cell_handle   Cell_handle;
+        typedef typename Tr::Facet         Facet;
+        typedef typename Tr::Edge          Edge;
+
+        typedef std::pair<Vertex_handle, Vertex_handle> Pair_of_vertices;
+
+        // computes and returns an ordered pair of Vertex
+        Pair_of_vertices
+        make_ordered_pair(const Vertex_handle vh1, const Vertex_handle vh2) const {
+          if (vh1 < vh2) {
+            return std::make_pair(vh1, vh2);
+          }
+          else {
+            return std::make_pair(vh2, vh1);
+          }
+        }
+        // same from an Edge
+        Pair_of_vertices
+        make_ordered_pair(const Edge e) const {
+          return make_ordered_pair(e.first->vertex(e.second),
+            e.first->vertex(e.third));
+        }
+        Facet canonical_facet(Cell_handle c, int i) const {
+          Cell_handle c2 = c->neighbor(i);
+          return (c2 < c) ? std::make_pair(c2, c2->index(c)) : std::make_pair(c, i);
+        }
+      }; // end class template C3t3_helper_class
+
+    } // end namespace TMDS_3::details
+  } //end namesapce TMDS_3
+
+/*!
+  \ingroup PkgTMDS3Classes
+
+   * @brief A data-structure to represent and maintain a 3D complex embedded
+ * in a 3D triangulation.
+
+  The class `Mesh_complex_3_in_triangulation_3` implements a data structure
+  to store the 3D restricted Delaunay triangulation used by a mesh
+  generation process.
+
+  This class is a model of the concept
+  `MeshComplexWithFeatures_3InTriangulation_3`.
+
+
+  \tparam Tr can be instantiated with any 3D
+  regular triangulation of \cgal provided that its
+  vertex and cell base class are models of the concepts
+  `MeshVertexBase_3` and `MeshCellBase_3`, respectively.
+
+  \tparam  CornerIndex is the type of the indices for corners. It must match the `Corner_index` of the model
+  of the `MeshDomainWithFeatures_3` concept used for mesh generation.
+
+  \tparam CurveIndex is the type of the indices for curves.
+  It must match the `Curve_index` types of the model
+  of the `MeshDomainWithFeatures_3` concept used for mesh generation.
+
+  Those two last template parameters defaults to `int`, so that they can be ignored
+  if the domain used for mesh generation does not include 0 and 1-dimensionnal features (i.e
+  is a model of the concept `MeshDomain_3`).
+
+  \cgalModels `MeshComplexWithFeatures_3InTriangulation_3`
+
+  \sa `CGAL::make_mesh_3()`
+  \sa `CGAL::refine_mesh_3()`
```

Comment:
In https://cgal.github.io/5693/v8/TMDS_3/classCGAL_1_1Mesh__complex__3__in__triangulation__3.html, the two first `See also` do not have links:
```
  \sa `CGAL::make_mesh_3()`
  \sa `CGAL::refine_mesh_3()`
```
but there was links in the old documentation: https://doc.cgal.org/latest/Mesh_3/classCGAL_1_1Mesh__complex__3__in__triangulation__3.html